iceberg-spark-runtime should not shade Hive classes it does not include in its shaded jar

I see the following exception when using spark-runtime jar. 
```
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/iceberg/shaded/org/apache/hadoop/hive/metastore/api/NoSuchObjectException
	at HiveTables$.<init>(HiveTables.scala:9)
	at HiveTables$.<clinit>(HiveTables.scala)
	at HiveTables.main(HiveTables.scala)
Caused by: java.lang.ClassNotFoundException: org.apache.iceberg.shaded.org.apache.hadoop.hive.metastore.api.NoSuchObjectException
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:338)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 3 more

```

It seems that we are shading Metastore classes even though we are not bundling them in the shaded jar. 

I have a patch working for this which I'll post later in the day.

--- a/build.gradle
+++ b/build.gradle
@@ -294,6 +294,9 @@ project(':iceberg-orc') {
     compile("org.apache.orc:orc-core::nohive") {
       exclude group: 'org.apache.hadoop'
       exclude group: 'commons-lang'
+      // These artifacts are shaded and included in the orc-core fat jar
+      exclude group: 'com.google.protobuf', module: 'protobuf-java'
+      exclude group: 'org.apache.hive', module: 'hive-storage-api'
     }
 
     compileOnly("org.apache.hadoop:hadoop-client") {
@@ -415,8 +418,6 @@ project(':iceberg-spark-runtime') {
     relocate 'org.codehaus.jackson', 'org.apache.iceberg.shaded.org.apache.parquet.shaded'
     relocate 'org.apache.orc', 'org.apache.iceberg.shaded.org.apache.orc'
     relocate 'io.airlift', 'org.apache.iceberg.shaded.io.airlift'
-    relocate 'org.apache.hive', 'org.apache.iceberg.shaded.org.apache.hive'
-    relocate 'org.apache.hadoop.hive', 'org.apache.iceberg.shaded.org.apache.hadoop.hive'
 
     archiveName = "iceberg-spark-runtime-${version}.${extension}"
   }

